### PR TITLE
[NBS1.0] Mark `backend-tasks` as deprecated

### PR DESCRIPTION
**Pull request by camilaibs**
## Hey, I just made a Pull Request!

Closes: https://github.com/backstage/backstage/issues/26047

Remove usage of `@backstage/backend-tasks` package in the main monorepo and mark it as deprecated.

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: packages/backend-tasks/src/index.ts [hunk 1]
```diff
@@ -17,7 +17,11 @@
...
  *
+ * @remarks
+ * This package is deprecated and will be removed in a near future.
+ * Please migrate to the new backend system, and depend on `coreServices.scheduler` from  `@backstage/backend-plugin-api` instead, or use `DefaultSchedulerService` from `@backstage/backend-defaults`.
+ *
  * @packageDocumentation
  */
 
-export * from './deprecated';
+export default 'This package is deprecated and will be removed in a near future. Please migrate to the new backend system, and depend on `coreServices.scheduler` from  `@backstage/backend-plugin-api` instead, or use `DefaultSchedulerService` from `@backstage/backend-defaults`.';
```
Comment (camilaibs) [inline packages/backend-tasks/src/index.ts:27]: @Rugvip If I didn't export something, the API report generation fails. Is it expected?

File: packages/backend-tasks/README.md
[the remark this answers is not in the document]
Comment (camilaibs) [reply packages/backend-tasks/README.md:1]: Done 🙏🏻

File: packages/backend-tasks/src/index.ts (hunk 1 above)
Comment (Rugvip) [reply packages/backend-tasks/src/index.ts:27]: Oh but hold on, we don't want to delete the contents of the package, just update the README + package.json 😅. If we remove the implementation we'll break the package for existing users.

Comment (camilaibs) [reply packages/backend-tasks/src/index.ts:27]: Just to make sure I understand, we can't delete this until we completely end support for the legacy system, right? I'll add the deleted `backend-tasks` files back.

Comment (camilaibs) [reply packages/backend-tasks/src/index.ts]: Files added back and package.json is marked as deprecated 🙏🏻

File: packages/backend-tasks/package.json [hunk 2]
```diff
@@ -51,8 +55,5 @@
     "@backstage/cli": "workspace:^",
     "wait-for-expect": "^3.0.2"
   },
-  "files": [
-    "dist",
-    "migrations/**/*.{js,d.ts}"
-  ]
+  "deprecated": true
```
Comment (Rugvip) [inline packages/backend-tasks/package.json:58]: Hmm, is this a thing? Wouldn't have expected a boolean. Where did you find it?

**Review by Rugvip — COMMENTED** [review]
Nice! 🎉 , just the `package.json` bit

File: packages/backend-tasks/package.json (hunk 2 above)
Comment (freben) [reply packages/backend-tasks/package.json:58]: oh is that what's successfully resulting in the bar at the top of https://www.npmjs.com/package/@backstage/plugin-scaffolder-backend-module-bitbucket ?

Comment (freben) [reply packages/backend-tasks/package.json:58]: Just seems it's not known [by node](https://nodejs.org/api/packages.html#nodejs-packagejson-field-definitions), nor [by npm](https://docs.npmjs.com/cli/v8/configuring-npm/package-json) 🤔

**Review by Rugvip — APPROVED** [review]

File: packages/backend-tasks/package.json (hunk 2 above)
Comment (camilaibs) [reply packages/backend-tasks/package.json]: Change reverted as we are going to configure that in the npm registry:
https://docs.npmjs.com/deprecating-and-undeprecating-packages-or-package-versions

**Pull request merged**